File: Tune/app/src/main/java/com/example/tune/activities/models/Post.java
package com.example.tune.activities.models;

import android.provider.MediaStore;

import java.io.Serializable;

public class Post implements Serializable {

    private String location;
    private User user;
    private MediaStore.Audio audio;
    private Genre genre;
    private Comment[] comments;

    public Post(String location, User user, MediaStore.Audio audio, Genre genre, Comment[] comments) {
        this.location = location;
        this.user = user;
        this.audio = audio;
        this.genre = genre;
        this.comments = comments;
    }

    public String getLocation() {
        return location;
    }

    public User getUser() {
        return user;
    }

    public MediaStore.Audio getAudio() {
        return audio;
    }

    public Genre getGenre() {
        return genre;
    }

    public Comment[] getComments() {
        return comments;
    }
}

File: tune_kotlin/app/src/main/java/com/example/tune_kotlin/models/Post.kt
package com.example.tune_kotlin.models

import java.io.File
import java.io.Serializable

class Post() :Serializable {
    var id: Long? = null
    var location: String? = null
    var email: String? = null
    var file: File? = null
    var filename: String? = null
    var genre: Genre? = null
    var comments: ArrayList<Comment>? = null
    var date: String? = null

    constructor(location: String, email: String?, filename: String, genre: Genre, date: String, comments: ArrayList<Comment>): this(){
        this.location = location
        this.email = email
        this. filename = filename
        this.genre = genre
        this.date = date
        this.comments = comments
    }

    constructor(location: String, email: String?, filename: String, genre: Genre, date: String): this(){
        this.location = location
        this.email = email
        this. filename = filename
        this.genre = genre
        this.date = date
    }


    override fun toString(): String {
        return "Post(id=$id, location=$location, email=$email, file=$file, filename=$filename, genre=$genre, comments=$comments, date=$date)"
    }

}